assembly/src/serde/mod.rs

```diff
@@ -33,22 +33,13 @@ impl ByteWriter {
         self.0.extend(val.to_le_bytes());
     }
 
-    pub fn write_len(&mut self, val: usize) -> Result<(), SerializationError> {
-        if val > u16::MAX as usize {
-            return Err(SerializationError::LengthTooLong);
-        }
-        self.write_u16(val as u16);
-        Ok(())
-    }
-
     pub fn write_felt(&mut self, val: Felt) {
         self.write_u64(val.as_int());
     }
 
-    pub fn write_str(&mut self, val: &str) -> Result<(), SerializationError> {
-        self.write_len(val.len())?;
+    pub fn write_str(&mut self, val: &str) {
+        self.write_u64(val.len() as u64);
         self.0.extend(val.as_bytes());
-        Ok(())
     }
```

Comment:
i would remove this method as winterfell `Serializable` trait doesn't have it - and so, we'd need to remove it anyway.

The way to handle this is to handle string serialization at the call sites of this method.